### PR TITLE
Add event API v2 server handler

### PR DESCRIPTION
**Pull request by MarlonGamez**
**Related**: #5422 , #5368 

**Description**
Adds `pkg/skaffold/server/v2` package which contains a new struct that handles the endpoints for the v2 API. Users will now be able to hit both the v1 and v2 endpoints through the running gRPC server.

Most of the server definition is copied from `pkg/skaffold/server` files, and the functions added to `pkg/skaffold/event/v2/event.go` are straight from `pkg/skaffold/event/event.go`

**Follow-up Work (remove if N/A)**
Emit protos on this server

### CODE REVIEW

**Comment by MarlonGamez** [comment]
Still need to add tests, but putting this up here in case anyone wants to take a look :)

File: pkg/skaffold/event/v2/event.go [hunk 1]
```diff
@@ -190,6 +241,19 @@ func emptyStatusCheckState() *proto.StatusCheckState {
 	}
 }
 
+func AutoTriggerDiff(name string, val bool) (bool, error) {
+	switch name {
+	case "build":
```
Comment (tejal29) [inline pkg/skaffold/event/v2/event.go:246]: we could reuse `pkg/errors.Phase` type and add `sync` to it?

**Review by tejal29 — APPROVED** [review]

File: pkg/skaffold/event/v2/event.go (hunk 1 above)
Comment (MarlonGamez) [reply pkg/skaffold/event/v2/event.go:246]: Good point! I'll update this

Comment (MarlonGamez) [reply pkg/skaffold/event/v2/event.go:246]: https://github.com/GoogleContainerTools/skaffold/pull/5625

**Pull request merged**